### PR TITLE
Add tests (using gtest)

### PR DESCRIPTION
**Pull request by mardy**
Show how to build unit tests using googletest.

### CODE REVIEW

**Review by DunderRoffe — APPROVED** [review]

**Comment by kursatkobya** [comment]
Approved with just two small comments,

- It  would be nice to add README file that this project uses googletest
- On test 19TemplatePublicClass looks a bit weird might create some questions. Maybe using decltype instead would make more sense.

File: tests/tst_library.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+#include "templatepublicclass.h"
+
+#include <gtest/gtest.h>
+#include <typeinfo>
+
+TEST(Library, class_instantiation)
+{
+    TemplatePublicClass obj;
+    std::string expectedType("19TemplatePublicClass");
```
Comment (kursatkobya) [inline tests/tst_library.cpp:20]: make sense to compare type itself instead of typename

**Review by kursatkobya — APPROVED** [review]

File: tests/tst_library.cpp (hunk 1 above)
Comment (mardy) [reply tests/tst_library.cpp:20]: Mmm... can you give me an example?

Comment (kursatkobya) [reply tests/tst_library.cpp:20]: yeah maybe decltype() comparison with TemplatePublicClass.
Or something like this
``EXPECT_TRUE((std::is_same<decltype(obj),TemplatePublicClass>::value));``

**Comment by mardy** [comment]
Updated. I added just one line to the README; a more extensive explanation is under work for inclusion in the SWF.

**Pull request merged**